packages/api-electron/main.js

```diff
@@ -17,49 +19,64 @@ program
 let win;
 
 function createWindow() {
-  const appJson = loadConfig();
-  if (appJson) {
+  loadConfig().then(appJson => {
     ssfElectron(appJson, program.symphony, program.developer);
-  } else {
+  }).catch(err => {
+    consoleError(err.message);
     process.exit();
-  }
+  });
 }
 
 function loadConfig() {
-  // the config json is passed to the ssf-electron bin script
-  const configFile = program.config ? process.cwd() + '/' + program.config : null;
-  const launchUrl = program.url;
-  if (configFile) {
-    if (fs.existsSync(configFile)) {
-      const config = JSON.parse(fs.readFileSync(configFile, 'utf8'));
+  if (program.config) {
+    return readConfigFile()
+      .then(data => new Promise((resolve, reject) => {
```

Comment:
Do you have to manually construct a Promise here?


```
   .then(data => new Promise((resolve, reject) => {
        const config = JSON.parse(data);

        if (program.url) {
          config.url = program.url;
        }

        if (config.url) {
          resolve(config);
        } else {
          reject(new Error('You must specify an URL (--url) or a config file containing an url (--config)'));
        }
      }));
```

could you instead throw rather than reject?

```
   .then(data => {
        const config = JSON.parse(data);

        if (program.url) {
          config.url = program.url;
        }

        if (config.url) {
          return config;
        } else {
          throw new Error('You must specify an URL (--url) or a config file containing an url (--config)');
        }
      });
```